src/workerd/jsg/ser.c++

```diff
@@ -7,8 +7,10 @@
 namespace workerd::jsg {
 
 Serializer::Serializer(Lock& js, kj::Maybe<Options> maybeOptions)
-    : isolate(js.v8Isolate),
-      ser(isolate, this) {
+    : ser(js.v8Isolate, this) {
+#ifdef KJ_DEBUG
+  kj::requireOnStack(this, "jsg::Serializer must be allocated on the stack");
```

Comment:
This seems quite subtle. Can this PR add a class comment flagging this constraint.

I'll see if we can add a DISALLOW_ALLOCATION macro to kj:
* https://cs.android.com/android/platform/superproject/main/+/main:art/libartbase/base/macros.h;drc=737b81f2bbf5e755abc51573dc2d5ef2de5fba90;l=41
* https://github.com/v8/v8/blob/d787388935ed0cb97fceb19ddc671f43df83b4a5/src/base/macros.h#L146